src/BlockTypes/ProductQuery.php

```diff
@@ -81,7 +81,7 @@ protected function initialize() {
 	 * @param array $parsed_block The block being rendered.
 	 * @return boolean
 	 */
-	private function is_woocommerce_variation( $parsed_block ) {
+	public static function is_woocommerce_variation( $parsed_block ) {
```

Comment:
Maybe this function should be moved to another class that could be shared between the 2 classes (ProductQuery and RelatedProducts)? Worth mentioning that PHP is not a skill that I'm strong at, so just wondering if we can avoid exposing and reusing a function that does not seems to be intrinsic to the ProductQuery class :thinking:

Reply:
I agree with you that it is not the best solution. I think we should refactor the ProductQuery class to be extendable by the variations. I didn't spend time on it because the design is not definitive, and we could change the implementation.

Furthermore, at least for now, we have just a variation that requires custom code on the PHP side, so doing this work now could increase the complexity unnecessarily.

What do you think?

Reply:
Ohh I see, considering that I think we can avoid increasing the complexity on this PR and change that when we get the final design (if needed)